ush/python/pygfs/task/nexus_emission.py

```diff
@@ -0,0 +1,486 @@
+        logger.info(f"NEXUS Logs: {self.task_config.DATA}/stderr")
+        logger.info(f"NEXUS Logs: {self.task_config.DATA}/NEXUS.log")
+
+        if not os.path.exists(self.task_config.DATA):
+            raise WorkflowException(f"Working directory does not exist: {self.task_config.DATA}")
+
+        # pprint(self.task_config)
+        exe = Executable(self.task_config.launcher)
+        arg_list = ['--ntasks',
+                    str(1),
+                    'nexus.x',
+                    '-c',
+                    self.task_config.NEXUS_CONFIG_NAME]
+        exe(*arg_list, output='stdout', error='stderr')
+
+        logger.info("Concatenating processed NEXUS files...")
+
+        # sort the files even though they should be sorted already | safety check
+        files = sorted(self.processed_nexus_files)
+
+        for i in files:
+
+            if not os.path.exists(i):
+                logger.warning(f"NEXUS file not found: {i}")
+                continue
+            else:
+                logger.info(f"NEXUS file found: {i}")
+
+        for f, dates in zip(files, self.forecast_dates):
+            logger.info(f" - {f}, {dates}")
+
+        # find the day indexes for each unique day
+        # this returns a dictionary
+        # example:
+        # {
+        #     datetime.date(2024, 1, 5): [0, 1, 3],
+        #     datetime.date(2024, 1, 6): [2]
+        # }
+        day_indexes = _get_day_indices(self.forecast_dates[:-1])  # hemco doesn't write out the last timestep
+        # now loop over each days
+        for date, indexes in day_indexes.items():
+            day_str = date.strftime('%Y%m%d')
+            logger.info(f"Processing NEXUS files for date: {date}")
+
+            dsets = []
+            print(indexes, len(files))
+            for index in indexes:
+                # list files for log
+                logger.info(f" - {files[index]}, {index}")
+
+                # now concatenate the files per day
+                if os.path.exists(files[index]) is False:
+                    break
+                ds = xr.open_dataset(files[index], decode_cf=False)
+
+                # update time coordinate
+                ds = ds.assign_coords(time=('time', [index]))
+
+                # set time units to reference start-date
+                ds.time.attrs['units'] = self.start_date.strftime('hours since %Y-%m-%d %H:00:00')
+
+                # append
+                dsets.append(ds)
+
+            # concatenate all the files for this day
+            if dsets is None:
+                break
+            else:
+                ds = xr.concat(dsets, dim='time')
+
+            encoding = {var: {"zlib": True, "complevel": 2} for var in ds.data_vars}
+            outname = f"{self.task_config.NEXUS_DIAG_PREFIX}.{day_str}.nc"
+            ds.to_netcdf(outname, format="NETCDF4", encoding=encoding)
+            logger.info(f"Wrote daily output: {outname}")
+
+        # dsets = []
```

Comment:
Either remove commented out lines or add a brief TODO explaining what still needs to be done